### DISCUSSION TITLE
How to display new image on the same place as the old image?

### DISCUSSION THREAD

**Discussion by GrayJack** [JuliaPlots/Makie.jl#1862] [Q&A]
I have a grid with parameters and a button to generate an image. After an image is created and the user changes a parameter and clicks on the bottom again, I want to delete the old image and display the new one, how can I do that?

I already tried to `delete!(figure, image_display)` followed by a new call to `image(figure[place], image)`, but that doesn't work for me, giving me the following:

```sh
Error in callback:
MethodError: no method matching delete!(::Figure, ::Makie.AxisPlot)
Closest candidates are:
  delete!(::Union{Attributes, AbstractPlot}, ::Any) at ~/.julia/packages/MakieCore/A0hGm/src/attributes.jl:163
  delete!(::Union{Figure, Scene}, ::DataInspector) at ~/.julia/packages/Makie/lgPZh/src/interaction/inspector.jl:391
  delete!(::DataStructures.RBTree{K}, ::K) where K at ~/.julia/packages/DataStructures/vSp4s/src/red_black_tree.jl:338
```

**Comment by asinghvi17** [comment]
You should use Observables to directly mutate the contents of the image!  The full docs can be found [here](https://makie.juliaplots.org/stable/documentation/nodes/), but some basic "pseudo" code is as follows:

```julia
f = Figure()
ax = Axis(fig[1, 1])
btn = Button(fig[2, 1]; tellwidth = false)

img_obs = Observable(rand(RGBAf, 10, 10))

img_plot = image!(ax, img_obs)
btn = Button(...)

on(btn.clicks) do n_clicks # triggers every time the button is pressed
    # next_image() must return a Matrix{RGBAf}
    img_obs[] = next_image() # update the input to `image!`
end
```
